subsys/lorawan/services/lorawan_services.c

```diff
@@ -160,6 +164,55 @@ int lorawan_services_reschedule_work(struct k_work_delayable *dwork, k_timeout_t
 	return k_work_reschedule_for_queue(&services_workq, dwork, delay);
 }
 
+int lorawan_services_class_c_start(void)
+{
+	int ret;
+
+	k_mutex_lock(&session_mutex, K_FOREVER);
+
+	if (active_class_c_sessions > 0) {
+		active_class_c_sessions++;
+		ret = active_class_c_sessions;
+	} else {
+		ret = lorawan_set_class(LORAWAN_CLASS_C);
+		if (ret == 0) {
+			LOG_DBG("Switched to class C");
+			active_class_c_sessions++;
+		}
+	}
+
+	k_mutex_unlock(&session_mutex);
+
+	return ret;
```

Comment:
If sticking with the mutex approach, this does not return the number of active sessions on the first call.

Reply:
Good spot. Fixed.